src/decoder/mod.rs

```diff
@@ -1055,6 +1055,19 @@ impl<R: Read + Seek> Decoder<R> {
         Ok(())
     }
 
+    /// TODO: this over-allocates when (bits_per_sample * samples) % 8 != 0.
+    /// If we leave in per-tile bit padding, then we cannot do it without knowing the number of tiles across this image is.
+    /// 
+    /// We have two options basically for precise allocation without knowing the number of tiles across:
+    ///  * Ban tiles where tile_dim % (bits_per_sample * samples) == 0
```

Comment:
Could you add a check if this condition isn't met, and return an unsupported error in that case?